### ISSUE TITLE
Package-Managed MSL 3.2.3 doesn't work under Linux

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#8800]
### Steps to Reproduce

1. Start with a fresh Linux user with empty .openmodelica directory, or delete the contents thereof.

2. Run the following script
```modelica
updatePackageIndex();getErrorString();
installPackage(Modelica, "3.2.3", exactMatch=false);getErrorString();
installPackage(Modelica, "4.0.0", exactMatch=false);getErrorString();
```
I get this output
```
true
"[/home/casella/.openmodelica/libraries/index.json:0:0-0:0:readonly]
Notification: Downloaded package index from URL https://libraries.openmodelica.org/index/v1/index.json.
"
true
"[/home/casella/.openmodelica/libraries/ModelicaServices 4.0.0+maint.om/package.mo:0:0-0:0:readonly] 
Notification: Package installed successfully (SHA c11123102e94d4b5acb139331a6d85f45d8adcc0).
[/home/casella/.openmodelica/libraries/Complex 4.0.0+maint.om/package.mo:0:0-0:0:readonly] 
Notification: Package installed successfully (SHA c11123102e94d4b5acb139331a6d85f45d8adcc0).
[/home/casella/.openmodelica/libraries/Modelica 3.2.3+maint.om/package.mo:0:0-0:0:readonly] 
Notification: Package installed successfully (SHA e68d7a0317c565ee00c1b8d44c527979ea0304bf).
"
true
"[/home/casella/.openmodelica/libraries/Modelica 4.0.0+maint.om/package.mo:0:0-0:0:readonly]
Notification: Package installed successfully (SHA c11123102e94d4b5acb139331a6d85f45d8adcc0).
"
```
I have no idea why ModelicaServices 4.0.0 and Complex 4.0.0 are installed first, while their 3.2.3 counterparts aren't, but for sure this causes trouble, see next point.

3. Run the following script
```modelica
loadString("
model M
  extends Modelica.Blocks.Examples.Filter;
  annotation(uses(Modelica(version=\"3.2.3\")));
end M;
");getErrorString();
simulate(M);getErrorString();
```
The following errors are obtained:
```
Notification: Automatically loaded package Complex 4.0.0 due to uses annotation.
[/home/casella/.openmodelica/libraries/ModelicaServices 4.0.0+maint.om/package.mo:202:5-202:11:writable] 
Error: No viable alternative near token: impure
Notification: Skipped loading package ModelicaServices (3.2.3,default) using 
MODELICAPATH /home/casella/.openmodelica/libraries/:/usr/bin/../lib/omlibrary (uses-annotation may be wrong).
[/home/casella/.openmodelica/libraries/Modelica 3.2.3+maint.om/Constants.mo:25:3-26:101:writable]
Error: Variable ModelicaServices.Machine.Integer_inf not found in scope Constants.
[/home/casella/.openmodelica/libraries/Modelica 3.2.3+maint.om/Blocks/Continuous.mo:2271:9-2271:37:writable]
Error: Import Modelica.Constants.pi not found in scope <top>.
[/home/casella/.openmodelica/libraries/Modelica 3.2.3+maint.om/Blocks/Continuous.mo:2293:11-2293:73:writable]
Error: Variable pi not found in scope Butterworth.
```

@sjoelund, can you please help? This is top priority for the next release.

**Comment by sjoelund** [comment]
It's not a problem with the package manager. Services 4.0.0 is compatible with MSL 3.2.3. However, the default language standard set when loading MSL 3.2.3 possibly prevents impure from being used.

**Comment by casella** [comment]
> It's not a problem with the package manager. Services 4.0.0 is compatible with MSL 3.2.3. 

Except that it uses language features that were not defined when MSL 3.2.3 was issued 😃 

>  However, the default language standard set when loading MSL 3.2.3 possibly prevents impure from being used.

Is there any compelling reason to set that standard when loading MSL 3.2.3?
- If the answer is yes, then Services 4.0.0 is not compatible with MSL 3.2.3 and we should act accordingly
- Otherwise, we should avoid setting that language standard

@sjoelund, @adrpo, what do you think?

**Comment by sjoelund** [comment]
The problem might be that some variable called impure might be used. It is compatible with all models in MSL 3.2.3 though. There would be no problem parsing it regardless.

**Comment by casella** [comment]
After discussion during the webmeeting, it looks like the best way to handle this issue is to have the parser (or lexer?) consider `impure` also for versions of Modelica < 3.4, possibly only if `--strict=false`.

This fix is slightly hackish, but it only affects old models using MSL 3.2.3, which are going to be obsolete in the long term, so that's no big deal.

@perost offered to take care of that.

**Comment by perost** [comment]
This should be fixed in #8803. Passing the `strict` flag to the parser was a bit more complicated than anticipated since changing the parser interface caused the bootstrapping to break, but I think it should work now.

**Comment by casella** [comment]
I tried my tests again and everything works fine now. Thanks!
